### PR TITLE
(feat): support for "begins_with" operator searches

### PR DESCRIPTION
**Pull request by sagarbaver**
I've added support for issuing queries using the "begins_with" operator.
This change partially contributes towards resolving https://github.com/aaronshaf/dynamodb-admin/issues/66 and improves user experience.

### CODE REVIEW

**Comment by sagarbaver** [comment]
@aaronshaf @rchl Please review. Thanks!

File: lib/backend.js [hunk 1]
```diff
@@ -450,11 +450,30 @@ exports.createServer = (dynamodb, docClient) => {
...
+                `#key${i} ${filters[key].operator} :key${i}`)
+            } else if (matchedKeySchema.KeyType === 'RANGE') {
+              if (filters[key].operator === 'begins_with') {
+                KeyConditionExpression.push(
+                  `${filters[key].operator} ( #key${i} , :key${i})`)
+              } else {
+                KeyConditionExpression.push(
+                  `#key${i} ${filters[key].operator} :key${i}`)
+              }
+            }
```
Comment (rchl) [inline lib/backend.js:467]: The code looks very repetitive. Can you clean this up a bit?

The `matchedKeySchema` variable can be initialized before the `if` block.
The `KeyConditionExpression.push` call can be done once after if/else checks if any filter is defined.

**Comment by rchl** [comment]
The select box that lists `begins_with` doesn't fit the text when selected:
Screenshot 2022-01-30 at 21 29 18

File: lib/backend.js [hunk 2]
```diff
@@ -441,20 +441,33 @@ exports.createServer = (dynamodb, docClient) => {
...
             filters[key].value = Number(filters[key].value)
           }
+
           ExpressionAttributeNames[`#key${i}`] = key
           ExpressionAttributeValues[`:key${i}`] = filters[key].value
-          if (indexBeingUsed && indexBeingUsed.KeySchema.find(
-            (keySchemaItem) => keySchemaItem.AttributeName === key)
-          ) {
-            KeyConditionExpression.push(`#key${i} ${filters[key].operator} :key${i}`)
+          const matchedKeySchema = indexBeingUsed !== undefined ? indexBeingUsed.KeySchema.find(
```
Comment (rchl) [inline lib/backend.js:452]: `indexBeingUsed` starts as `null` so this needs to be:
```suggestion
          const matchedKeySchema = indexBeingUsed !== null ? indexBeingUsed.KeySchema.find(
```
or just a truthy check like before.

**Review by rchl — CHANGES_REQUESTED** [review]

**Review by rchl — APPROVED** [review]

**Comment by rchl** [comment]
thanks!

**Comment by sagarbaver** [comment]
@rchl @aaronshaf Could one of you please make these changes available on Docker Hub?
https://hub.docker.com/r/aaronshaf/dynamodb-admin/tags

**Comment by rchl** [comment]
Updated

**Pull request merged**